docs/config_file.md

```diff
@@ -0,0 +1,304 @@
+
+The `cross` configuration in the `Cross.toml` file can contain the following
+elements:
+
+
+# `build`
+
+The `build` key allows you to set global variables, e.g.:
+
+> *NOTE*: `$CROSS_DEB_ARCH` is automatically provided by cross, [see
+> here][custom_images_automatic_arch].
+
+```toml
+[build]
+build-std = false                              # do not build the std library. has precedence over xargo
+xargo = true                                   # enable the use of xargo by default
+zig = false                                    # do not use zig cc for the builds
+default-target = "x86_64-unknown-linux-gnu"    # use this target if none is explicitly provided
+pre-build = [                                  # additional commands to run prior to building the package
+    "dpkg --add-architecture $CROSS_DEB_ARCH", 
+    "apt-get update && apt-get --assume-yes install libssl-dev:$CROSS_DEB_ARCH"
+]                 
+```
+
+
+# `build.env`
+
+With the `build.env` key you can globally set volumes that should be mounted in
+the Docker container or environment variables that should be passed through.
+For example:
+
+```toml
+[build.env]
+volumes = ["VOL1_ARG", "VOL2_ARG=/path/to/volume"]
+passthrough = ["VAR1_ARG", "VAR2_ARG=VALUE"]
+```
+
+Note how in the environment variable passthrough, we can provide a definition
+for the variable as well. `VAR1_ARG` will be the value of the environment
+variable on the host, while `VAR2_ARG` will be `VALUE`. Likewise, the path to
+the volume for `VOL1_ARG` will be the value of the environment variable on the
+host, while `VOL2_ARG` will be `/path/to/volume`.
+
+
+# `build.dockerfile`
+
+> If the image you want to use is already available from a container registry,
+> check out the `target.TARGET.image` option below.
+
+The `build.dockerfile` key lets you provide a custom Docker image for all
+targets, except those specified `target.TARGET.dockerfile`. The value can be
+provided as either a table or a string. If `build.dockerfile` is set to a
+string, it's equivalent to setting `build.dockerfile.file` to that value. For
+example, using only a string:
+
+```toml
+[build]
+dockerfile = "./Dockerfile"
+```
+
+Or using a table:
+
+```toml
+[build.dockerfile]
+file = "./Dockerfile"         # the dockerfile to use relative to the `Cargo.toml`
+context = "."                 # the context folder to build the script in. defaults to `.`
+build-args = { ARG1 = "foo" } # https://docs.docker.com/engine/reference/builder/#arg
+```
+
+`cross` will build and use the image that was built instead of the default
+image. It's recommended to base your custom image on the default Docker image
+that `cross` uses: `ghcr.io/cross-rs/{{TARGET}}:{{VERSION}}` (where
+`{{VERSION}}` is `cross`'s version). This way you won't have to figure out how
+to install a cross-C toolchain in your custom image.
+
+> *NOTE*: `$CROSS_DEB_ARCH` is automatically provided by cross, [see
+> here][custom_images_automatic_arch].
```

Comment:
github ui is giving me red marks in the document from here and onwards. strange

![image](https://github.com/cross-rs/cross/assets/1502855/b8133430-a3d2-4e21-9a5e-bfd67c6cb9db)

I can't see anything obviously wrong with it, doubt this is a problem with the file and just how they parse md for syntax

Reply:
I have these marks, too, but the rendered HTML is fine. I think it's safe to ignore this.